akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
```

Comment:
This method will be called whenever a name server resolution is requested and the caching time has been expired that it's configured by default to 120s. See at the end of default [configuration](http://doc.akka.io/docs/akka/current/general/configuration.html#akka-actor)